apps/deploy-web/src/components/sdl/GpuFormControl.tsx

```diff
@@ -43,6 +43,33 @@ export const GpuFormControl: React.FunctionComponent<Props> = ({ gpuModels, cont
     appendFormGpuModel({ vendor: "nvidia", name: "", memory: "", interface: "" });
   };
 
+  const gpuVendors = useMemo(() => {
+    return gpuModels
+      ? gpuModels.map((vendor, index) => {
+          return { id: index, value: vendor.name };
+        })
+      : hardCodedGpuVendors;
+  }, [gpuModels]);
+
+  const gpuModelsByName: Record<string, GpuModel> = useMemo(() => {
```

Comment:
non-blocking: some code could be dropped.

```suggestion
import keyBy from "lodash/keyBy";
// ....
const gpuModelsByName: Record<string, GpuModel> = keyBy(gpuModels, "name");
```

Reply:
It is a bit more complicated than that, because gpuModels is an array of GpuVendors:

```
Array<{
  models: Array<{
    name: string;
  }>
]>
```

But you're right about we can simplify it with lodash functions, I just did.

Reply:
oh, I did miss that. new code looks cleaner indeed though, thx